src/features/channels/components/channel-list/modal-hidden-list/ModalHiddenList.tsx

```diff
@@ -0,0 +1,87 @@
+import Modal from '@metis/common/components/ui/Modal';
+import { useAppDispatch, useAppSelector } from '@metis/store/hooks';
+import { openToast } from '@metis/store/ui/ui.slice';
+import IndeterminateCheckBoxIcon from '@mui/icons-material/IndeterminateCheckBox';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import { ListItem, ListItemButton, ListItemIcon, ListItemText, Typography } from '@mui/material';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import Divider from '@mui/material/Divider';
+import * as React from 'react';
+import { useMemo } from 'react';
+import { selectState, unhideChannel } from '../../../store/channel.slice';
+import ChannelListItem from '../channel-list-item/ChannelListItem';
+import useStyles from './ModalHiddenList.styles';
+
+const ModalHiddenList = () => {
+  const styles = useStyles();
+  const [open, setOpen] = React.useState(false);
+  const dispatch = useAppDispatch();
+  const { channels, hiddenChannels } = useAppSelector(selectState);
+  const hiddenChannelsAddreses = useMemo(
+    () => hiddenChannels.map((channel) => channel.channelAddress),
+    [hiddenChannels]
+  );
+  const closeModal = () => {
+    setOpen(false);
+  };
+  const showChannel = (channelAddress: string, channelName: string) => {
+    dispatch(unhideChannel(channelAddress));
+    dispatch(
+      openToast({
+        type: 'info',
+        text: `${channelName} was successfully removed from Hidden Channels`,
+      })
+    );
+  };
+  return (
+    <>
+      <ListItem disablePadding>
+        <ListItemButton onClick={() => setOpen(true)}>
+          <ListItemIcon style={{ color: 'purple' }}>
+            <IndeterminateCheckBoxIcon />
+          </ListItemIcon>
+          <ListItemText primary="Hidden Channels" />
+        </ListItemButton>
+      </ListItem>
```

Comment:
Do not put this 'ListItem' here, it'll be hard to find it later... Just leave the modal here and pass the 'open' and 'onClose' properties as props

Reply:
Ok